spec/app/lib/bing_api_v7_spec.rb

```diff
@@ -1,15 +1,59 @@
 require 'rails_helper'
-require 'bing_api_v7'
-
-describe 'BingApiV7' do
-  # let('search') { nil }
-  bing_response = BingApiV7.new.search('anything')
-  it 'search = nil or empty' do      
-    expect(bing_response).not_to be_empty
-    expect(bing_response).to be_an_instance_of(Hash)
-    expect(bing_response).to include("webPages")
-    expect(bing_response["webPages"]).to include("value")
-    expect(bing_response["webPages"]["value"]).to be_a_kind_of(Hash)
+require 'json'
+require 'webmock/rspec'
+require 'support/vcr_setup'
+
+RSpec.configure do |config|
+  # Add VCR to all tests
+  config.around(:each) do |example|
+    options = example.metadata[:vcr] || {}
+    if options[:record] == :skip 
+      VCR.turned_off(&example)
+    else
+      name = example.metadata[:full_description].split(/\s+/, 2).join('/').underscore.gsub(/\./,'/').gsub(/[^\w\/]+/, '_').gsub(/\/$/, '')
+      VCR.use_cassette(name, options, &example)
+    end
   end
+end 
+
+RSpec.describe BingApiV7 do
   
+  describe '#search', vcr: true do
+
+    subject { BingApiV7.new }
+
+    # # let(:http) { double }
+
+
+    # # before(:each) do
+    # #   allow(http).to receive(:request).with(an_instance_of(Net::HTTP::Get)).and_return(response)
+    # #   allow(Net::HTTP).to receive(:start).and_yield(http)
+    # # end
+
+    context 'with valid response data' do
+    #   # let(:response) { { 'webPages' => { 'value' => { name: 'CompanyName' } } }.to_json }
+      let(:data) { subject.search('motorola') } 
```

Comment:
1. use let! instead of let
2. speak with teammate who was working on FullContact. He has a code sample how to stub external calls.

Reply:
`let` is not a correct place for this case!
`let` should be used to store a data, not a subject because it executes not within an example context but in a hooks